src/api/spec/models/package_spec.rb

```diff
@@ -98,4 +104,74 @@
       end
     end
   end
+
+  context '#maintainers' do
+    before do
+      User.current = user
+    end
+
+    it 'returns an array with user objects to all maintainers for a package' do
+      # first of all, we add a user who is not a maintainer but a bugowner
+      # he/she should not be recognized by package.maintainers
+      package.add_user(other_user2, Role.find_by_title!("bugowner"))
```

Comment:
Instead of doing:
```ruby
package.add_user(other_user2, Role.find_by_title!("bugowner"))
package.save!
```

Just create and use  a factory `relationship_package_user_as_bugowner ` somehow as @bgeuken said:
```ruby
create(:relationship_package_user_as_bugowner, user: other_user2, package: package)
```
Here you have the factory:
```ruby
# In the file src/api/spec/factories/relationship.rb
...
factory :relationship_package_user do
  package
  user

  factory :relationship_package_user_as_bugowner do
    role { Role.find_by_title('bugowner') }
  end
end
...
```